### ISSUE TITLE
perl "panic"ed

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#5273]
Migrated from [rt.perl.org#8862](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=8862) (status was 'resolved')

Searchable as RT8862$

**Comment by p5pRT** [comment]
### From goldbb2@earthlink.net

#### Created by goldbb2@earthlink.net



\[Windows 95\] C&#8203;:\\>perl \-c
length $\_\[0\] ?
&nbsp;   $foo\->bar\($\_\[0\]\) :
&nbsp;   $state = 3\, $foo\->baz\, &$blat;
\_\_END\_\_
panic&#8203;: unexpected optree near method call at \- line 3\, near "3\,"
\- had compilation errors\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl v5.6.1:

Configured by gsar at Wed Jan  2 17:16:07 2002.

Summary of my perl5 (revision 5 version 6 subversion 1) configuration:
  Platform:
    osname=MSWin32, osvers=4.0, archname=MSWin32-x86-multi-thread
    uname=''
    config_args='undef'
    hint=recommended, useposix=true, d_sigaction=undef
    usethreads=undef use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=undef d_sfio=undef uselargefiles=undef usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
  Compiler:
    cc='cl', ccflags ='-nologo -O1 -MD -DNDEBUG -DWIN32 -D_CONSOLE -DNO_STRICT -DHAVE_DES_FCRYPT  -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DPERL_MSVCRT_READFIX',
    optimize='-O1 -MD -DNDEBUG',
    cppflags='-DWIN32'
    ccversion='', gccversion='', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=undef, longlongsize=8, d_longdbl=define, longdblsize=10
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=8, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='link', ldflags ='-nologo -nodefaultlib -release  -libpath:"C:\Ben\ActivePerl\Perl\lib\CORE"  -machine:x86'
    libpth=C:\DEVSTU~2\VC98\lib
    libs=  oldnames.lib kernel32.lib user32.lib gdi32.lib winspool.lib  comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib  netapi32.lib uuid.lib wsock32.lib mpr.lib winmm.lib  version.lib odbc32.lib odbccp32.lib msvcrt.lib
    perllibs=  oldnames.lib kernel32.lib user32.lib gdi32.lib winspool.lib  comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib  netapi32.lib uuid.lib wsock32.lib mpr.lib winmm.lib  version.lib odbc32.lib odbccp32.lib msvcrt.lib
    libc=msvcrt.lib, so=dll, useshrplib=yes, libperl=perl56.lib
  Dynamic Linking:
    dlsrc=dl_win32.xs, dlext=dll, d_dlsymun=undef, ccdlflags=' '
    cccdlflags=' ', lddlflags='-dll -nologo -nodefaultlib -release  -libpath:"C:\Ben\ActivePerl\Perl\lib\CORE"  -machine:x86'

Locally applied patches:
    ACTIVEPERL_LOCAL_PATCHES_ENTRY


@INC for perl v5.6.1:
    C:/Ben/ActivePerl/Perl/lib
    C:/Ben/ActivePerl/Perl/site/lib
    .


Environment for perl v5.6.1:
    HOME (unset)
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=C:\WINDOWS;C:\WINDOWS\COMMAND;C:\BEN\ACTIVEPERL\PERL\BIN
    PERL_BADLANG (unset)
    SHELL (unset)


```
</details>

**Comment by p5pRT** [comment]
### From @jhi

On Wed\, Mar 20\, 2002 at 09&#8203;:58&#8203;:18PM \-0500\, Benjamin Goldberg wrote&#8203;:
> This is a bug report for perl from goldbb2@&#8203;earthlink\.net\,
> generated with the help of perlbug 1\.33 running under perl v5\.6\.1\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> \[Windows 95\] C&#8203;:\\>perl \-c
> length $\_\[0\] ?
>    $foo\->bar\($\_\[0\]\) :
>    $state = 3\, $foo\->baz\, &$blat;
> \_\_END\_\_
> panic&#8203;: unexpected optree near method call at \- line 3\, near "3\,"
> \- had compilation errors\.

With bleadperl&#8203;:

ugli&#8203;:/tmp/jhi/perl ; cat >\! x
length $\_\[0\] ?
&nbsp;   $foo\->bar\($\_\[0\]\) :
&nbsp;   $state = 3\, $foo\->baz\, &$blat;
ugli&#8203;:/tmp/jhi/perl ; \./perl \-Ilib x
Can't call method "baz" on an undefined value at x line 1\.
ugli&#8203;:/tmp/jhi/perl ; 
 
\-\- 
$jhi\+\+; \# http&#8203;://www\.iki\.fi/jhi/
&nbsp;        \# There is this special biologist word we use for 'stable'\.
&nbsp;        \# It is 'dead'\. \-\- Jack Cohen

**Comment by p5pRT** [comment]
### From @schwern

Real bug in 5\.6\.1\.  Fixed in bleadperl\.

\-\- 

Michael G\. Schwern   \<schwern@&#8203;pobox\.com>    http&#8203;://www\.pobox\.com/~schwern/
Perl Quality Assurance        \<perl\-qa@&#8203;perl\.org>           Kwalitee Is Job One
I have this god\-awful need to aquire useless crap\!\!\!

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

Jarkko Hietaniemi wrote&#8203;:
> 
> On Wed\, Mar 20\, 2002 at 09&#8203;:58&#8203;:18PM \-0500\, Benjamin Goldberg wrote&#8203;:
> > This is a bug report for perl from goldbb2@&#8203;earthlink\.net\,
> > generated with the help of perlbug 1\.33 running under perl v5\.6\.1\.
> >
> >
> > \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> > \[Please enter your report here\]
> >
> > \[Windows 95\] C&#8203;:\\>perl \-c
> > length $\_\[0\] ?
> >    $foo\->bar\($\_\[0\]\) :
> >    $state = 3\, $foo\->baz\, &$blat;
> > \_\_END\_\_
> > panic&#8203;: unexpected optree near method call at \- line 3\, near "3\,"
> > \- had compilation errors\.
> 
> With bleadperl&#8203;:
> 
> ugli&#8203;:/tmp/jhi/perl ; cat >\! x
> length $\_\[0\] ?
>    $foo\->bar\($\_\[0\]\) :
>    $state = 3\, $foo\->baz\, &$blat;
> ugli&#8203;:/tmp/jhi/perl ; \./perl \-Ilib x
> Can't call method "baz" on an undefined value at x line 1\.
> ugli&#8203;:/tmp/jhi/perl ;

I'm glad it's gone\, but I wonder what caused it\.\.\.

If it was fixed intentionally\, \(and more importantly\, if we have a test
for this bug which passes following patch xyzzy but fails for prior
versions\)\, good\, but if it was fixed by accident\, then maybe we should
add this or something like it as a regression test?

\-\- 
print reverse\( "\,rekcah"\, " lreP"\, " rehtona"\, " tsuJ" \)\."\\n";

**Comment by p5pRT** [comment]
### From @rgarcia

On 2002\.03\.21 06&#8203;:10 Benjamin Goldberg wrote&#8203;:
> > With bleadperl&#8203;:
> > 
> > ugli&#8203;:/tmp/jhi/perl ; cat >\! x
> > length $\_\[0\] ?
> >    $foo\->bar\($\_\[0\]\) :
> >    $state = 3\, $foo\->baz\, &$blat;
> > ugli&#8203;:/tmp/jhi/perl ; \./perl \-Ilib x
> > Can't call method "baz" on an undefined value at x line 1\.
> > ugli&#8203;:/tmp/jhi/perl ;
> 
> I'm glad it's gone\, but I wonder what caused it\.\.\.
> 
> If it was fixed intentionally\, \(and more importantly\, if we have a test
> for this bug which passes following patch xyzzy but fails for prior
> versions\)\, good\, but if it was fixed by accident\, then maybe we should
> add this or something like it as a regression test?

If I'm not mistaken\, this has been fixed intentionally by patch \#13404\, and
a test was included\.
